select.c

```diff
@@ -177,12 +178,30 @@ select_dispatch(struct event_base *base, struct timeval *tv)
 	check_selectop(sop);
 
 	if (res == -1) {
-		if (errno != EINTR) {
-			event_warn("select");
-			return (-1);
+		if (errno == EINTR) {
+		        return (0);
 		}
-
-		return (0);
+                /* There seems to be a very subtle race condition between the
```

Comment:
Hm, interesting, any chance that this can be covered with the unit test?

Reply:
I'm really sorry for abandoning this, sadly, but I haven't have enough time for libevent for a long time already.

But I'm trying to prepare the release and so looking for the pending PRs

So I've started with a test, it is not correct, but maybe it will give you an idea on how to make it right - https://github.com/azat/libevent/commit/b49b3ad80225877eb26ba2649c9ff3f6838b1124

But then I get back to the code and I don't see a problem here, in case of `select` fails the loop will break and return an error (i.e. from `event_base_loop`), can you elaborate this?